source/robot_model/src/Model.cpp

```diff
@@ -299,23 +299,23 @@ Model::inverse_kinematics(const state_representation::CartesianPose& cartesian_p
   const std::chrono::nanoseconds dt(static_cast<int>(1e9));
   // initialization of the loop variables
   state_representation::JointPositions q(joint_positions);
-  state_representation::JointVelocities dq(this->get_robot_name(), this->get_number_of_joints());
-  state_representation::Jacobian J(this->get_robot_name(), this->robot_model_.nq, frame_name);
-  Eigen::MatrixXd J_b = J.data();
+  state_representation::JointVelocities dq(this->get_robot_name(), joint_positions.get_names());
+  state_representation::Jacobian J = this->compute_jacobian(joint_positions, frame_name);
+  Eigen::MatrixXd J_b =  Eigen::MatrixXd::Zero(6, this->get_number_of_joints());
   Eigen::MatrixXd JJt(6, 6);
-  Eigen::MatrixXd W_b = Eigen::MatrixXd::Identity(this->robot_model_.nq, this->robot_model_.nq);
-  Eigen::MatrixXd W_c = Eigen::MatrixXd::Identity(this->robot_model_.nq, this->robot_model_.nq);
-  Eigen::VectorXd psi(this->robot_model_.nq);
+  Eigen::MatrixXd W_b = Eigen::MatrixXd::Identity(this->get_number_of_joints(), this->get_number_of_joints());
+  Eigen::MatrixXd W_c = Eigen::MatrixXd::Identity(this->get_number_of_joints(), this->get_number_of_joints());
+  Eigen::VectorXd psi(this->get_number_of_joints());
   Eigen::VectorXd err(6);
   for (unsigned int i = 0; i < parameters.max_number_of_iterations; ++i) {
     err = ((cartesian_pose - this->forward_kinematics(q, frame_id)) / dt).data();
     // break in case of convergence
     if (err.cwiseAbs().maxCoeff() < parameters.tolerance) {
       return q;
     }
-    J = this->compute_jacobian(q);
+    J = this->compute_jacobian(q, frame_name);
```

Comment:
Oh yes I forgot to mention here the frame name was missing and line 303 I also compute the `Jacobian` using the same function. We don't have a test to actually check whether this IK works for different frames than the end-effector one but for sure without this change it would not have worked anyway.